### DISCUSSION TITLE
Can I use `computed` inside a Pinia store?

### DISCUSSION THREAD

**Discussion by rodrigocfd** [vuejs/pinia#1765] [Help and Questions] [answered]
When I need a computed value from the store, can I use `computed` and export it?

Example:

```typescript
import {computed, reactive} from 'vue';
import {defineStore} from 'pinia';

const useFoo = defineStore('foo', () => {
    const state = reactive({
        name: 'Joe',
        age: 41,
    });

    const prefixed = computed(() => `Mr. ${state.name}`);
    const withAge = computed(() => `${state.name}, ${state.age}`);

    function setName(name: string): void {
        state.name = name;
    }

    return {prefixed, withAge, setName};
});
```

If this approach is not correct, how could I write it?

**Comment by posva** [comment, marked as answer]
Yes, [that's exactly how to define a getter](https://pinia.vuejs.org/core-concepts/#setup-stores)

note you also need to return the state for it be picked up by devtools and, more importantly, for SSR:

```ts
return {
  ...toRefs(state),
  prefixed,
  withAge,
  // you don't need a setName action if it just sets the name
}
```

Also worth seeing: https://pinia.vuejs.org/core-concepts/state.html#resetting-the-state

**Reply by rodrigocfd** [reply]
That's great, thank you.
